ext/aes256gcm_decrypt/aes256gcm_decrypt.c

```diff
@@ -43,55 +50,60 @@ VALUE method_aes256gcm_decrypt_decrypt(VALUE self, VALUE rb_ciphertext, VALUE rb
 
   /* Initialise decryption operation */
   if (!EVP_DecryptInit_ex(ctx, EVP_aes_256_gcm(), NULL, NULL, NULL)) {
-    goto cleanup1;
+    goto cleanup3;
   }
 
   /* Set initialisation vector (IV) length */
   if (!EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_GCM_SET_IVLEN, iv_len, NULL)) {
-    goto cleanup1;
+    goto cleanup3;
   }
 
   /* Initialise key and IV */
   if (!EVP_DecryptInit_ex(ctx, NULL, NULL, key, iv)) {
-    goto cleanup1;
+    goto cleanup3;
   }
 
   /* Provide empty string as additional authenticated data (AAD) */
-  if (!EVP_DecryptUpdate(ctx, NULL, &len, "", 0)) {
-    goto cleanup1;
+  if (!EVP_DecryptUpdate(ctx, NULL, &len, (unsigned char *)"", 0)) {
+    goto cleanup3;
   }
```

Comment:
I don't think this call should be included. It is superfluous according to [this wiki page](https://wiki.openssl.org/index.php/EVP_Authenticated_Encryption_and_Decryption#Authenticated_Decryption_using_GCM_mode). It seems like it just adds complexity without any benefit.

Reply:
Well spotted! :+1: